### PR TITLE
feat(mirror): fix apt mirror and refactor code

### PR DESCRIPTION
**Pull request by paologallinaharbur**
This PR:
 - fix mirroring
 - refactor code a bit to unify loops
 - adds timeout for commands

### CODE REVIEW

File: publisher/publisher.go [hunk 1]
```diff
@@ -36,19 +38,20 @@ const (
...
+	defaultAptlyFolder = "/root/.aptly"
+	aptPoolMain        = "pool/main/"
+	aptDists           = "dists/"
+	commandTimeout     = time.Hour * 1
 )
 
 var (
 	l                = log.New(log.Writer(), "", 0)
-	streamExecOutput = true
+	streamExecOutput = false
```
Comment (varas) [inline publisher/publisher.go:54]: I think we can enable this

File: publisher/publisher.go [hunk 2]
```diff
@@ -174,64 +177,100 @@ func parseUploadSchema(fileContent []byte) (uploadArtifactsSchema, error) {
...
 	return nil
 }
 
 func downloadArtifacts(conf config, schema uploadArtifactsSchema) error {
 	for _, artifactSchema := range schema {
-		err := downloadArtifact(conf, artifactSchema)
-		if err != nil {
-			return err
+		for _, arch := range artifactSchema.Arch {
+			err := downloadArtifact(conf, artifactSchema.Src, arch)
```
Comment (varas) [inline publisher/publisher.go:235]: 👏

File: publisher/publisher.go [hunk 3]
```diff
@@ -280,7 +319,7 @@ func uploadRpm(conf config, srcTemplate string, upload Upload, arch string) (err
 
 		l.Printf("Waiting for file creation")
```
Comment (varas) [inline publisher/publisher.go:320]: s/Waiting/Checking ?

**Review by varas — APPROVED** [review]

File: publisher/publisher.go (hunk 1 above)
Comment (paologallinaharbur) [reply publisher/publisher.go:54]: done

File: publisher/publisher.go (hunk 3 above)
Comment (paologallinaharbur) [reply publisher/publisher.go:320]: removed old comment

**Pull request merged**